packages/k6-tests/scripts/E2E.js

```diff
@@ -0,0 +1,277 @@
+      clearTimersParties();
+      wsParties.close();
+
+      const quoteId = crypto.randomUUID();
+      const transactionId = crypto.randomUUID();
+      const wsChannelQuotes = `${traceParent.traceId}/PUT/quotes/${quoteId}`;
+      const wsURLQuotes = `${wsUrl}/${wsChannelQuotes}`
+      const wsQuotes = new WebSocket(wsURLQuotes);
+
+      var wsTimeoutId = null;
+
+      const clearTimersQuotes = () => {
+        if (wsTimeoutId) { clearTimeout(wsTimeoutId); wsTimeoutId=null }
+      }
+
+      wsQuotes.onclose(() => {
+        clearTimersQuotes();
+      });
+
+      wsQuotes.onerror((err) => {
+        console.error(traceId, err);
+        check(err, { 'QUOTES_E2E_FSPIOP_POST_QUOTES_SUCCESS': (cbMessage) => false });
+        clearTimersQuotes();
+        wsQuotes.close();
+      });
+
+      wsQuotes.onmessage = (event) => {
+        console.info(traceId, `WS message received [${wsChannelQuotes}]: ${event.data}`);
+        check(event.data, { 'QUOTES_E2E_FSPIOP_POST_QUOTES_SUCCESS': (cbMessage) => cbMessage == 'SUCCESS_CALLBACK_RECEIVED' });
+        clearTimersQuotes();
+        wsQuotes.close();
+
+        const transferId = crypto.randomUUID();
+        const wsChannelTransfers = `${traceParent.traceId}/PUT/transfers/${transferId}`;
+        const wsURLTransfers = `${wsUrl}/${wsChannelTransfers}`
+        const wsTransfers = new WebSocket(wsURLTransfers);
+
+        var wsTimeoutId = null;
+
+        const clearTimersTransfers = () => {
+          if (wsTimeoutId) { clearTimeout(wsTimeoutId); wsTimeoutId=null }
+        }
+
+        wsTransfers.onclose(() => {
+          clearTimersTransfers();
+        });
+
+        wsTransfers.onerror((err) => {
+          console.error(traceId, err);
+          check(err, { 'TRANSFERS_E2E_FSPIOP_POST_TRANSFERS_SUCCESS': (cbMessage) => false });
+          clearTimersTransfers();
+          wsTransfers.close();
+        });
+
+        wsTransfers.onmessage = (event) => {
+          console.info(traceId, `WS message received [${wsChannelTransfers}]: ${event.data}`);
+          check(event.data, { 'TRANSFERS_E2E_FSPIOP_POST_TRANSFERS_SUCCESS': (cbMessage) => cbMessage == 'SUCCESS_CALLBACK_RECEIVED' });
+          clearTimersTransfers();
+          wsTransfers.close();
+        };
+
+        wsTransfers.onopen = () => {
+          console.info(traceId, `WS open on URL: ${wsUrl}`);
+          const params = {
+            tags: {
+              payerFspId,
+              payeeFspId
+            },
+            headers: {
+              'Accept': 'application/vnd.interoperability.transfers+json;version=1.1',
+              'Content-Type': 'application/vnd.interoperability.transfers+json;version=1.1',
+              'FSPIOP-Source': payerFspId,
+              'FSPIOP-Destination': payeeFspId,
+              'Date': (new Date()).toUTCString(),
+              'traceparent': traceParent.toString(),
+              'tracestate': `tx_end2end_start_ts=${startTs}`
```

Comment:
The calculation of startTs feels wrong to me.
We need to calculate it in each phase or we can have separate timestamp variables appended at each phase like tx_end2end_parties_start_ts, tx_end2end_quotes_start_ts...etc

Reply:
done